cmd/modelcmd/apicontext.go

```diff
@@ -17,11 +19,19 @@ import (
 	"github.com/juju/juju/jujuclient"
 )
 
+type CookieJar interface {
+	http.CookieJar
+	RemoveAllHost(host string)
+}
+
 // APIContext holds the context required for making connections to
 // APIs used by juju.
 type APIContext struct {
-	Jar            *cookiejar.Jar
-	WebPageVisitor httpbakery.Visitor
+	Jar CookieJar
+	// jar holds the internal version of the cookie jar - it has
+	// methods that clients should not use, such as Save.
+	jar            *domainCookieJar
```

Comment:
OK, I've removed the Jar field and replaced it with a CookieJar method that returns the internal jar implementation, so there's no possibility of them getting out of step.